docker/docker.go

```diff
@@ -345,19 +374,31 @@ func WaitForContainer(ctx context.Context, dClient *client.Client, hostname stri
 	if dClient == nil {
 		return 1, fmt.Errorf("Failed waiting for container: docker client is nil for container [%s] on host [%s]", containerName, hostname)
 	}
-	// We capture the status exit code of the container
-	statusCh, errCh := dClient.ContainerWait(ctx, containerName, container.WaitConditionNotRunning)
-	select {
-	case err := <-errCh:
+	// 5 minutes timeout, especially for transferring snapshots
+	for retries := 0; retries < 300; retries++ {
```

Comment:
I am not sure we should rewrite this part of docker client, I can see example of setting timeout to ContainerWait function https://godoc.org/github.com/docker/docker/client#example-Client-ContainerWait-WithTimeout